### DISCUSSION TITLE
Draft for a better general UI structure

### DISCUSSION THREAD

**Discussion by math-GH** [FreshRSS/FreshRSS#5966] [Ideas]
(Follow up from https://github.com/FreshRSS/FreshRSS/pull/5907#issuecomment-1826835083 )

Here is a draft of my idea how to improve the UI of FreshRSS.

The goal is to have a better handling with all "backend" features.

Let me show the draft:

## main view (desktop)

![grafik](https://github.com/FreshRSS/FreshRSS/assets/1645099/393c422d-aef2-4f60-ad8f-6fafe8f344eb)
1. The search + config menu panel moves from top to the left side. 
 - Plus button: add a new feed or category (this button moves from the feed menu to here)
 - the button with the FreshRSS logo: sub menu for the subscription management  (this button moves from the feed menu to here)
 - the view buttons (list view, global view, reader view) move from the tool bar to here. 
 - help: sub menu for about FreshRSS and manual
 - person: sub menu for the personel account settings incl. the logout button
 - cog: sub menu for the system admin configs (only visible for admins)
2. the text search form moves from the panel above to here
3. only the article list will scroll. The scrolling will do not having any effect to the feed list and the new navigation on the left hand side. The tool bar (2) will be fixed.
4. the navigation buttons flow over the article list

### subscription management
![grafik](https://github.com/FreshRSS/FreshRSS/assets/1645099/3525df4f-1a43-4ebb-b5df-e07c510f7127)

### help
![grafik](https://github.com/FreshRSS/FreshRSS/assets/1645099/7ee1740d-6f4d-4be3-b66e-0de1adb79b94)

### account
![grafik](https://github.com/FreshRSS/FreshRSS/assets/1645099/7977a9f5-28ea-47dd-a325-3ae69e6694bb)

### system admin
![grafik](https://github.com/FreshRSS/FreshRSS/assets/1645099/d8d2842a-508f-41a1-a001-ba761fa5974b)

## mobile view
![grafik](https://github.com/FreshRSS/FreshRSS/assets/1645099/20bf84be-0be7-4ab0-b167-de2d1f92165e)

1. the new top navigation panel is on top of the page. The button that opens the side panel moved here (from the tool bar below)
2. the tool bar will be fixed while scrolling

What do you think? Please give me feedback

**Comment by Alkarex** [comment]
Looks good at a quick glance 👍🏻  I will try to come back to it tomorrow. 
1. If some of https://github.com/FreshRSS/FreshRSS/discussions/5900 could be incorporated, it would be even better (i.e. add control of density horizontally and vertically - we already have most of what we need already, based on mobile view and the existing display options, so there is not much coding needed, besides UI controls)
2. Some day, I would like to have a way to select one or more articles to apply actions (and by the way, when the horizontal density needs to be increased, actions such as read and star could be hidden to be solely in that actions menu)

**Reply by math-GH** [reply]
Thanks for this feedback. I would like to improve this points later. I want to focus on the navigation here first.

**Reply by math-GH** [reply]
@Alkarex Do you have any more comments, ideas, improvements, or suggestions?

**Reply by Alkarex** [reply]
I like the idea to refresh a bit the UI, and the mobile version in particular is very in need of improvement :-)
So good for me 👍🏻 
At the moment, I do not have more concrete feedback than my two points above, but I would also like to keep an eye on the performance (e.g. during scroll with many articles), also on older hardware.

**Comment by beerisgood** [comment]
Looks good but i would like to slim the mobile view. Too much buttons with little to none space between.  Also the navigation buttons at bottom aren’t needed in my opinion and looks more modern without. 

On main view side „1“ use too much space. Best would be an hiding/open this section with an button

**Reply by Frenzie** [reply]
>  Also the navigation buttons at bottom aren’t needed in my opinion and looks more modern without.

I use them all the time. On mobile that is, rarely if ever on laptop/desktop.

One could do without relatively easily, see EasyRSS, but I'd prefer to call that replacing the functionality.

**Reply by math-GH** [reply]
Thanks for this feedback. Yes you are right, the buttons need some better styling. 

Regarding the navigation buttons:
Please check the `Display` configuration: 
![grafik](https://github.com/FreshRSS/FreshRSS/assets/1645099/a9a089c8-509b-47c7-a200-47a00ef1d090)

**Reply by math-GH** [reply]
buttons better styled in the mobile view:
![grafik](https://github.com/FreshRSS/FreshRSS/assets/1645099/22efa17f-d0a2-4a63-84bb-9a3dd10411a8)

**Comment by math-GH** [comment]
First implementation is ready for have a first look: #6047
